src/services/photo/index.js

```diff
@@ -49,73 +68,118 @@ function sendUploadToGCS (req, res, next) {
+          return Promise.resolve({
+            url: filePublicUrl,
+            mimetype: mimetype,
+            size: size
+          });
+        });
+      });
+  }
+}
+
 module.exports = function(){
   const app = this;
   //TODO(A): Need id and need to support multiple photos uplaod
   //TODO(A): Also need to support photo url and download it instead of 3rd party app
-  app.post('/photos',
-      uploader.single('image'),
-      sendUploadToGCS,
-      function(req, res, next) {
-        // we don't care req.body for now.
-        // (care only file content)
-        var imageUrl;
-        if (req.file && req.file.cloudStoragePublicUrl) {
-          imageUrl = req.file.cloudStoragePublicUrl;
-        }
-        res.json({
-          url: imageUrl,
-          mimetype: req.file.mimetype,
-          size: req.file.size
-        });
-      });
+  app.use('/photos', prepareMultipart, attachFileToFeathers, new PhotosService());
 
   // This service receives image url. Then, it downloads and stores image for you.
   // TODO(A): support downloading multiple urls
-  app.post('/photos/uploadfromurl',
-      function (req, res, next) {
-        const data = req.body;
-        var photoUrls = [];
-        console.log(data);
-        if (Array.isArray(data)) {
-          photoUrls = data;
-          return res.send(new errors.NotImplemented('Array is not supported yet'));
-        } else {
-          photoUrls.push(data);
-        }
-        const url = photoUrls[0];
-        // TODO(A): Change to promise and ES6 style
-        request
-          .head(url)
-          .end(function (err, photoHeaderResp) {
-            if (err) {
-              return res.send(err);
-            }
-            const pathArray = urlparser.parse(url).pathname.split('/');
-            const filename = pathArray[pathArray.length - 1];
-            const mimetype = photoHeaderResp.header['content-type'];
-            const size = photoHeaderResp.header['content-length'];
-            const gcsname = Date.now() + '_' + filename;
-            const gcsfile = bucket.file(gcsname);
-            const filePublicUrl = getPublicUrl(gcsname);
-            console.log('Downloading photo...');
-            console.log('Name: ' + filename);
-            console.log('Mimetype: ' + mimetype);
-            console.log('Size: ' + size);
-            console.log('To: ' + filePublicUrl);
-            var uploadPipe = request.get(url).pipe(gcsfile.createWriteStream());
-            uploadPipe.on('error', function(err) {
-              console.log(err);
-              res.send(err);
-            });
-            uploadPipe.on('finish', function() {
-              console.log('Uploading to the cloud starge is complete!');
-              res.json({
-                url: filePublicUrl,
-                mimetype: mimetype,
-                size: size
-              });
-            });
-          });
-      });
+  app.use('/photos/upload_from_url', new UploadPhotoFromUrlService());
```

Comment:
That's alright. No choice :P. You can merge this PR and fix it in another PR as a separate issue.